kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/components/menus/EphemeralSelectMenu.kt

```diff
@@ -19,19 +18,16 @@ import com.kotlindiscord.kord.extensions.utils.scheduling.Task
 import dev.kord.common.annotation.KordUnsafe
 import dev.kord.core.behavior.interaction.modal
 import dev.kord.core.behavior.interaction.respondEphemeral
+import dev.kord.core.behavior.interaction.response.EphemeralMessageInteractionResponseBehavior
 import dev.kord.core.event.interaction.SelectMenuInteractionCreateEvent
-import dev.kord.rest.builder.message.create.InteractionResponseCreateBuilder
-
-public typealias InitialEphemeralSelectMenuResponseBuilder =
-    (suspend InteractionResponseCreateBuilder.(SelectMenuInteractionCreateEvent) -> Unit)?
 
-/** Class representing an ephemeral-only select (dropdown) menu. **/
-public open class EphemeralSelectMenu<M : ModalForm>(
+/** Interface for ephemeral select menus. **/
+public abstract class EphemeralSelectMenu<C, M : ModalForm>(
```

Comment:
Since `C` needs to meet 2 conditions, it is defined in the where clause a few lines below (`where C : SelectMenuContext, C : EphemeralInteractionContext`)